libyul/AsmPrinter.cpp

```diff
@@ -258,6 +268,32 @@ string AsmPrinter::operator()(Block const& _block) const
 	}
 }
 
+string AsmPrinter::operator()(Comment const& _comment) const
+{
+	auto const static trimWhitespace = [](string _s) -> string
+	{
+		auto const static isnospace = [](int _ch) { return !std::isspace(_ch); };
+		_s.erase(_s.begin(), std::find_if(_s.begin(), _s.end(), isnospace));
+		_s.erase(find_if(_s.rbegin(), _s.rend(), isnospace).base(), _s.end());
+		return _s;
+	};
+	auto const trimmedText = trimWhitespace(_comment.text);
+
+	vector<string> lines;
+	boost::split(lines, trimmedText, [](auto _ch) -> bool { return _ch == '\n'; });
+
+	stringstream out;
+
+	if (m_blockStatementIndex > 0)
+		out << '\n';        // One leading newline for better readability of comments.
```

Comment:
Ah I see. Isn't it a bit much code just to prevent adding an empty line at the beginning of a block?

Reply:
Maybe it is much less code if you just use a bool for that?